### PR TITLE
R: split tex output for use with texlive.combine

### LINKED ISSUE #21952: R's Sweave.sty not accessible to texlive
## Issue description

The R package in nixpkgs produces the directory `lib/R/share/texmf/tex/latex/` which has files like `Sweave.sty`. These files (to my knowledge) serve only to allow LaTeX to use the `.tex` files which Sweave produces - but texlive can't exactly find them.

I would solve this myself but I don't yet know Nix well enough. If anyone could point to any packages that solve a similar problem, that would help.

### Steps to reproduce

Install `R` and `texlive.combined.scheme-full` (other texlive packages may work similarly). Save the below as `example.Rnw`:

```latex
\documentclass{article}
\begin{document}
<<>>=
print("foo")
@ 
\end{document}
```

Then run:

    R CMD Sweave example.Rnw
    latex example.tex

latex will halt with:

    ! LaTeX Error: File `Sweave.sty' not found.

This has a somewhat kludgey workaround in the form of instead running something like:

    TEXINPUTS=".:/nix/store/i8cbhc7q0l1lkq3ynzxx0q5dh4s3dmd1-R-3.2.3/lib/R/share/texmf/tex/latex:" latex example.tex

## Technical details

* System: 16.09.1324.1dd0fb6 (Flounder)
* Nix version: nix-env (Nix) 1.11.4
* Nixpkgs version: "16.09.1324.1dd0fb6"

### PR DESCRIPTION
**Pull request by xworld21**
## Description of changes
Fix #21952

## Things done
- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by omasanori** [comment]
5 packages marked as broken and skipped:
spark spark3 spark_3_2 spark_3_3 spark_3_4

4 packages failed to build:
python310Packages.cnvkit python310Packages.cnvkit.dist python311Packages.cnvkit python311Packages.cnvkit.dist

48 packages built:
R R.tex cantor diffoscope diffoscope.dist diffoscope.man golden-cheetah-bin haskellPackages.H haskellPackages.H.doc haskellPackages.ihaskell-inline-r haskellPackages.ihaskell-inline-r.doc haskellPackages.inline-r haskellPackages.inline-r.doc labplot postgresql11JitPackages.plr postgresql11Packages.plr postgresql12JitPackages.plr postgresql12Packages.plr postgresql13JitPackages.plr postgresql13Packages.plr postgresqlJitPackages.plr postgresqlPackages.plr postgresql15JitPackages.plr postgresql15Packages.plr python310Packages.nbdev python310Packages.nbdev.dist python310Packages.radian python310Packages.radian.dist python310Packages.rchitect python310Packages.rchitect.dist python310Packages.rpy2 python310Packages.rpy2.dist python311Packages.nbdev python311Packages.nbdev.dist python311Packages.radian python311Packages.radian.dist python311Packages.rchitect python311Packages.rchitect.dist python311Packages.rpy2 python311Packages.rpy2.dist quarto rstudio rstudio-server rstudioServerWrapper rstudioWrapper sage sageWithDoc tests.pkg-config.defaultPkgConfigPackages.libR

File: pkgs/applications/science/math/R/default.nix [hunk 1]
```diff
@@ -89,6 +91,13 @@ stdenv.mkDerivation (finalAttrs: {
 
   installTargets = [ "install" "install-info" "install-pdf" ];
 
+  # move tex files to $tex for use with texlive.combine
+  # add link in $out since ${R_SHARE_DIR}/texmf is hardcoded in several places
+  postInstall = ''
+    mv -T "$out/lib/R/share/texmf" "$tex"
```
Comment (omasanori) [inline pkgs/applications/science/math/R/default.nix:97]: Would you make a subdirectory like `$tex/R` and put things into it?

File: pkgs/applications/science/math/R/default.nix [hunk 2]
```diff
@@ -103,6 +112,10 @@ stdenv.mkDerivation (finalAttrs: {
 
   passthru.tests.pkg-config = testers.testMetaPkgConfig finalAttrs.finalPackage;
 
+  # make tex output available to texlive.combine
+  passthru.tlType = "run";
```
Comment (omasanori) [inline pkgs/applications/science/math/R/default.nix:116]: I am not so sure but isn't `tlDeps` needed?

**Review by omasanori — COMMENTED** [review]
To be clear, what I should test is:

1. `nix-shell -p 'texlive.combine { inherit (texlive) scheme-small; inherit R; }'`
2. Steps to reproduce in https://github.com/NixOS/nixpkgs/issues/21952

right? I have not yet succeeded but I have some minor suggestions.

File: pkgs/applications/science/math/R/default.nix (hunk 2 above)
Comment (xworld21) [reply pkgs/applications/science/math/R/default.nix]: Good point, I have added the packages used in `Sweave.sty` and similar as dependencies.

File: pkgs/applications/science/math/R/default.nix (hunk 1 above)
Comment (xworld21) [reply pkgs/applications/science/math/R/default.nix:97]: R already uses the "TeX Directory Structure", so the files are already in the correct location (which is `tex/latex` for the packages, `bibtex/...` for the bibtex stuff).

Comment (omasanori) [reply pkgs/applications/science/math/R/default.nix:97]: OK. Thanks for clarification.

**Comment by xworld21** [comment]
> 1. `nix-shell -p 'texlive.combine { inherit (texlive) scheme-small; inherit R; }'`
> 2. Steps to reproduce in [R's Sweave.sty not accessible to texlive #21952](https://github.com/NixOS/nixpkgs/issues/21952)

You also need to include R itself, i.e.
```
nix-shell -p R 'texlive.combine { inherit (texlive) scheme-small; inherit R; }'
```
The `inherit R;` only combines the tex packages.

**Comment by omasanori** [comment]
Thanks for your guidance.

**Review by omasanori — APPROVED** [review]
I confirmed that this fixes the problem with the following steps:

1. `nix-channel --add https://github.com/xworld21/nixpkgs/archive/refs/heads/R-sweave-tex.tar.gz nixpkgs`
2. `nix-channel --update`
3. `nix-shell -p R 'texlive.combine { inherit (texlive) scheme-small; inherit R; }'`
4. Check if `kpsewhere Sweave.sty` finds a path to Sweave.sty
5. Check if the steps in https://github.com/NixOS/nixpkgs/issues/21952 succeeds

**Pull request merged**